pkg/daemon/gpsd.go

```diff
@@ -126,7 +126,7 @@ func (g *GPSD) CmdStop() {
 		}
 	}
 	g.unRegisterSubscriber()
-	<-g.exitCh
+	<-g.exitCh // waiting for all child routines to exit; we could add timeout to avoid waiting
```

Comment:
Is it possible that MonitorGNSSEventsWithUblox will consume from this channel and then this will hang forever?

Reply:
yes possible ; when one function reads from a single channel consumed by two functions, the other function will indeed hang waiting if monitoring is not  waiting on this channel. I will modify